src/main/java/com/amazonaws/encryptionsdk/kms/AwsKmsCmkId.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk.kms;
+
+import com.amazonaws.arn.Arn;
+import com.amazonaws.encryptionsdk.exception.MalformedArnException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+
+import java.util.Objects;
+
+/**
+ * A representation of an AWS KMS Customer Master Key Identifier, which may be one either a
+ * key ID, key Amazon Resource Name (ARN), alias name, or alias ARN.
+ */
+public class AwsKmsCmkId {
+
+    private static final String ARN_PREFIX = "arn:";
+    private String keyId;
+
+    private AwsKmsCmkId(String keyId) throws MalformedArnException {
+        Validate.notBlank(keyId, "keyId must be a valid CMK Amazon Resource Name (ARN), CMK Alias, Key Id or Alias ARN");
```

Comment:
I think you'd be a lot more helpful to state explicitly "keyId cannot be blank"

Reply:
I'll go with "keyId is required" to match the rest of the validation code

Reply:
I think that meets the minimum bar. Is there a reason we can't be more specific and say "don't provide a blank string"? I'm in favour of helping out developers that have hooked things up incorrectly as specifically as possible.

Reply:
Validate.notBlank is checking if the string is null, whitespace only, empty string etc, which I'm not sure is widely recognized as the definition of "blank". I'm not sure I see how saying don't provide a blank string is actually more specific than saying it is required.

Reply:
I could say "keyId must be neither null, a length of zero (no characters), empty nor whitespace"

Reply:
Validate.notBlank is checking if the string is null, whitespace only, empty string etc, which I'm not sure is widely recognized as the definition of "blank". How about I say "keyId must be neither null, a length of zero (no characters), empty nor whitespace"

Reply:
Yes I think that's reasonable, however you want to word it. I'm just imagining someone hitting this and getting "keyId is required" and saying "but I AM passing it dammit, and I know it can't be null..." :)